File: java/com/epicdragonworld/gameserver/network/packets/receivable/ObjectInfoRequest.java
package com.epicdragonworld.gameserver.network.packets.receivable;

import com.epicdragonworld.gameserver.managers.WorldManager;
import com.epicdragonworld.gameserver.model.WorldObject;
import com.epicdragonworld.gameserver.model.actor.Player;
import com.epicdragonworld.gameserver.network.GameClient;
import com.epicdragonworld.gameserver.network.ReceivablePacket;
import com.epicdragonworld.gameserver.network.packets.sendable.PlayerInformation;

/**
 * @author Pantelis Andrianakis
 */
public class ObjectInfoRequest
{
	public ObjectInfoRequest(GameClient client, ReceivablePacket packet)
	{
		// Read data.
		final long objectId = packet.readLong();
		
		// Get the acting player.
		final Player player = client.getActiveChar();
		// Send the information.
		for (WorldObject object : WorldManager.getInstance().getVisibleObjects(player))
		{
			if (object.getObjectId() == objectId)
			{
				if (object.isPlayer())
				{
					// Send the information to the current player.
					final Player otherPlayer = (Player) object;
					client.channelSend(new PlayerInformation(otherPlayer));
					// Send information to the other player as well.
					final PlayerInformation playerInfo = new PlayerInformation(player);
					otherPlayer.channelSend(playerInfo);
				}
				// TODO: Other objects.
				break;
			}
		}
	}
}

File: EpicDragonWorldServer/Network/Packets/Receivable/ObjectInfoRequest.cs
using System.Threading.Tasks;

/**
 * Author: Pantelis Andrianakis
 * Date: November 7th 2018
 */
public class ObjectInfoRequest
{
    public ObjectInfoRequest(GameClient client, ReceivablePacket packet)
    {
        // Read data.
        long objectId = packet.ReadLong();

        // Get the acting player.
        Player player = client.GetActiveChar();
        // Send the information.
        foreach (WorldObject obj in WorldManager.GetVisibleObjects(player))
        {
            if (obj.GetObjectId() == objectId)
            {
                if (obj.IsPlayer())
                {
                    client.ChannelSend(new PlayerInformation(obj.AsPlayer()));
                }
                // TODO: Other objects - NpcInformation?

                // Send delayed animation update in case object was already moving.
                Task.Delay(1000).ContinueWith(task => SendAnimationInfo(client, obj));
                break;
            }
        }
    }

    private void SendAnimationInfo(GameClient client, WorldObject obj)
    {
        if (obj != null)
        {
            AnimationHolder animations = obj.GetAnimations();
            if (animations != null)
            {
                client.ChannelSend(new AnimatorUpdate(obj.GetObjectId(), animations.velocityX, animations.velocityZ, animations.triggerJump, animations.isInWater, animations.isGrounded));
            }
        }
    }
}
